Player/src/main/java/com/github/sef24sp4/player/SpeedControl.java

```diff
@@ -95,6 +81,6 @@ public void updateSpeed(long currentTime) {
 	 * @return The entity's speed when moving diagonally.
 	 */
 	public double getDiagonalSpeed() {
-		return this.speed * (this.speed / (Math.sqrt(2 * (this.speed * this.speed))));
+		return this.getSpeed() * (this.getSpeed() / (Math.sqrt(2 * (this.getSpeed() * this.getSpeed()))));
```

Comment:
Save result of `getSpeed()` into local final variable to avoid multiple calls to method.